src/qutip_tensornetwork/testing.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+def assert_network_close(actual, desired, atol=0, rtol=1e-7):
```

Comment:
I agree that we should relly in other packages for testing isomorphism. I am not sure if networkx is the right tool though as it seems to be quite slow compared to other packages: https://graph-tool.skewed.de/performance.  I must admit that have not looked too much into the other packages either so it may be possible that they are not flexible enough. Lets keep this for another PR although I will add an issue to remember this needs to be changed.

Reply:
I think the page you linked to says why networkx is a good fit for our case -- our graphs will be comparatively small and its pure Python. We should probably check that the isomorphism is fast for reasonably sized networks though -- e.g. maybe a few thousand nodes -- if we're going to use it outside of tests. Probably calling back into custom node_match and edge_match functions from C is slow and a pain though, so not sure if graph-tool will support that or perform well on it.